torchgeo/datasets/geo.py

```diff
@@ -375,6 +378,20 @@ def __init__(
                 f"No {self.__class__.__name__} data was found in '{root}'"
             )
 
+        if not self.all_bands:
+            band_indexes = None
+        else:
+            if self.bands:
+                band_indexes = [self.all_bands.index(i) + 1 for i in self.bands]
+                assert len(band_indexes) == len(self.bands)
+            else:
+                band_indexes = None
+
+            if self.rgb_bands:
+                rgb_band_indexes = [self.all_bands.index(i) + 1 for i in self.rgb_bands]
+                assert len(rgb_band_indexes) == len(self.rgb_bands)
```

Comment:
This variable doesn't seem to be used anywhere. I think it will have the side-effect that you aren't allowed to not include all RGB bands during model training, which we don't want. I would remove this if-statement.

Reply:
Removed